Packages/com.unity.inputsystem/InputSystem/Plugins/OnScreen/OnScreenControl.cs

```diff
@@ -299,5 +299,30 @@ public void Destroy()
         }
 
         private static InlinedArray<OnScreenDeviceInfo> s_OnScreenDevices;
+
+        internal string GetWarningMessage()
+        {
+            return $"{GetType()} needs to be attached as a child to a UI Canvas and have a RectTransform component to function properly.";
```

Comment:
Minor: a property would be sufficient (e.g. string warningMessage => $"{GetType()} needs t.....";)